src/util/type.h

```diff
@@ -64,6 +63,15 @@ class typet:public irept
     return static_cast<typet &>(sub.front());
   }
 
+public:
+  typet &add_subtype()
```

Comment:
As discussed out of band: it would be nice to have a comment that explains when (not) to use this.

Reply:
Done